### DISCUSSION TITLE
NiftiSaver

### DISCUSSION THREAD

**Discussion by udiram** [Project-MONAI/tutorials#1494] [Q&A]
**Describe the bug**
while trying to save niftis, error message is unclear and no clear solution, other than a rollback, appears viable

**To Reproduce**
import and use NiftiSaver on monai 1.2.0

**Expected behavior**
information on where NifiSaver is now imported from, or whether it is deprecated. 

**Environment (please complete the following information):**
 - OS: ubuntu 22.04
MONAI version: 1.2.0
Numpy version: 1.25.2
Pytorch version: 2.0.1+cu117
MONAI flags: HAS_EXT = False, USE_COMPILED = False, USE_META_DICT = False
MONAI rev id: c33f1ba588ee00229a309000e888f9817b4f1934
MONAI __file__: /home/exouser/.local/lib/python3.10/site-packages/monai/__init__.py

Optional dependencies:
Pytorch Ignite version: 0.4.11
ITK version: 5.3.0
Nibabel version: 5.1.0
scikit-image version: 0.21.0
Pillow version: 9.0.1
Tensorboard version: 2.14.0
gdown version: 4.7.1
TorchVision version: 0.15.2+cu117
tqdm version: 4.66.1
lmdb version: 1.4.1
psutil version: 5.9.0
pandas version: 2.0.3
einops version: 0.6.1
transformers version: 4.21.3
mlflow version: 2.6.0
pynrrd version: 1.0.0

Python 3.10.12
Driver Version: 525.85.05    CUDA Version: 12.0
GRID A100X-40C

**Additional context**
worked previously (1 or 2 releases ago), but fails to provide adequate resolution in user facing warnings/error.

**Comment by KumoLiu** [comment]
Hi @udiram, thanks for your suggestion here.
I think we have removed this API since v0.9, perhaps you could use `NibabelWriter` now.
https://github.com/Project-MONAI/MONAI/blob/dev/CHANGELOG.md#removed

Hope it helps, thanks!

**Comment by udiram** [comment]
hi  @KumoLiu , thanks for your response! I am able to save it by rolling back my monai to the last working version I had, 1.0.0, but keeping other packages up to date like numpy, to avoid the "missing image reader for file .nii.gz" issue. I was wondering if there are any monai frameworks that I could use to reverse the transforms that i load the images in with in val_loader, because the images I am inferencing on are then being shifted/cropped and I would like to overlay these, which would mean reversing the original training/validation transforms after inference. 

thanks,

**Comment by KumoLiu** [comment]
Hi @udiram, you can use inverse to invert your data.
For more details, you can refer to this tutorial.
https://github.com/Project-MONAI/tutorials/blob/main/modules/inverse_transforms_and_test_time_augmentations.ipynb

Or you can check the "Evaluation on original image spacings" section in most of the end-to-end tutorials.
https://github.com/Project-MONAI/tutorials/blob/main/3d_segmentation/brats_segmentation_3d.ipynb

Hope it helps, thanks!
